lib/webbluetooth/bindings.js

```diff
@@ -0,0 +1,399 @@
+'use strict';
+
+var util = require('util');
+var events = require('events');
+
+var debug = require('debug')('webble-bindings');
+
+function makeList(uuid){
+  return {services:[ uuid ]};
+}
+
+function addDashes(uuid){
+  if(!uuid || typeof uuid !== 'string'){
+    return uuid;
+  }
+  if(uuid && uuid.length === 32){
+    uuid = uuid.substring(0,8) + '-' + uuid.substring(8,12) + '-' +  uuid.substring(12,16) + '-' +  uuid.substring(16,20) + '-' +  uuid.substring(20);
+  }
+  return uuid.toLowerCase();
+}
+
+function stripDashes(uuid){
+  if(typeof uuid === 'string'){
+    uuid = uuid.split('-').join('');
+  }
+  return uuid;
+}
+
+
+var NobleBindings = function() {
+
+  this._ble = null;
+  this._startScanCommand = null;
+  this._peripherals = {};
+
+  var self = this;
+
+};
+util.inherits(NobleBindings, events.EventEmitter);
+
+NobleBindings.prototype.init = function(ble) {
+
+  if(ble) {
+    this._ble = ble;
+  }else {
+    this._ble = navigator.bluetooth;
+  }
+
+  var self = this;
+  process.nextTick(function(){
+    debug('initing');
+    if(!self._ble){
+      return self.emit('error', new Error('This browser does not support WebBluetooth.'));
+    }
+    console.log('emit powered on');
+    self.emit('stateChange', 'poweredOn');
+  });
+};
+
+
+NobleBindings.prototype.onOpen = function() {
+  debug('on -> open');
+};
+
+NobleBindings.prototype.onClose = function() {
+  debug('on -> close');
+
+  this.emit('stateChange', 'poweredOff');
+};
+
+NobleBindings.prototype.startScanning = function(options, allowDuplicates) {
+  var self = this;
+
+  if(Array.isArray(options)){
+    options = {services: options};
+  }
+
+  if(typeof options !== 'object'){
+    options = {services: options};
+  }
+
+  if(!Array.isArray(options.services)){
+    options.services = [options.services];
+  }
+
+  options.services = options.services.map(function(service){
+    //web bluetooth requires 4 char hex service names to be passed in as integers
+    if(typeof service === 'string' && service.length === 4){
+      service = parseInt('0x' + service);
+    }
+    else if(typeof service === 'string' && service.length === 6 && service.indexOf('0x') === 0){
+      service = parseInt(service);
+    }
+    return service;
+  });
+
+  var dashedUuids = options.services.map(addDashes);
+
+  var filterList = dashedUuids.map(makeList);
+  if(options.name){
+    filterList.push({name: options.name});
+  }
+  if(options.namePrefix){
+    filterList.push({namePrefix: options.namePrefix});
+  }
+
+  var request = {filters: filterList};
+
+  debug('startScanning', request, allowDuplicates);
+
+  this._ble.requestDevice(request)
+    .then(function(device){
+      debug('scan finished', device);
+      self.emit('scanStop', {});
+      if(device){
+
+        var address = device.id;
+        //TODO use device.adData when api is ready
+        //rssi = device.adData.rssi;
```

Comment:
It will be part of the [new scanning API](https://webbluetoothcg.github.io/web-bluetooth/scanning.html)  .  Right now I'm just faking scanning with `requestDevice()`